course-definition.yml

```diff
@@ -557,3 +565,273 @@ stages:
       **Optional:** To improve download speeds, you can download from multiple peers at once. You could have a work queue consisting of each piece that needs to be downloaded. Your worker (connection with a peer) could pick a piece from the work queue, attempt to download it, check the integrity, and write the downloaded piece into a buffer. Any failure (network issue, hashes not matching, peer not having the piece etc.) would put the piece back into the work queue to be tried again.
     marketing_md: |-
       In this stage, you'll download the entire file. You'll download all the pieces, verify them using SHA-1 and save them to disk.
+
+  - slug: "mg1"
+    primary_extension_slug: "magnet-links"
+    name: "Parse Magnet Link"
+    difficulty: easy
+    description_md: |-
+    Welcome to the BitTorrent Magnet Links extension! In this extension, you'll add support for downloading files using magnet links.
+
+    Magnet links are hyperlinks that allow users to download files from peers without needing a torrent file.
+
+    They contain minimal information necessary to discover peers. Clients can then download metadata that would be found in a .torrent file from peers.
+
+    Magnet links follow this format:\
+    magnet:?xt=urn:btih:*\<info-hash\>*&dn=*\<name\>*&tr=*\<tracker-url\>*&x.pe=*\<peer-address\>*
```

Comment:
I wonder if I should omit `x.pe` parameter to simplify, we don't use it anywhere, but added it for completeness since it's in the spec

Reply:
Yep, would remove - not only do we not use it, it doesn't seem to be present in magnet links generated by transmission either.